do-task-now.lua

```diff
@@ -0,0 +1,226 @@
+        )
+    then
+        doJobNow(building.jobs[0])
+        return
+    end
+    print("Couldn't find any job for " .. utils.getBuildingName(building))
+end
+
+local function doUnitJobNow(unit)
+    if dfhack.units.isCitizen(unit) then
+        --print('This will attempt to make a job of ' .. getUnitName(unit) .. ' a top priority')
+        local t_job = unit.job
+        if t_job then
+            local job = t_job.current_job
+            if job then
+                doJobNow(job)
+                return
+            end
+        end
+        print("Couldn't find any job for " .. getUnitName(unit) )
+    else
+        --print('This will attempt to make a job with ' .. getUnitName(unit) .. ' a top priority')
+
+        local needle = unit.id
+        for _link, job in utils.listpairs(df.global.world.jobs.list) do
+            if #job.general_refs > 0 then
+                for _, gref in ipairs(job.general_refs) do
+                    --if gref:getType() == df.general_ref_type.UNIT then -- can't do this: there are many different types that can be units
+                    local u = gref:getUnit()
+                    if u and u.id == needle then
+                        doJobNow(job)
+                        return
+                    end
+                    --end
+                end
+            end
+        end
+
+        print("Couldn't find any job involving " .. getUnitName(unit) )
+    end
+end
+
+local function doPlantJobNow(plant)
+    --print('This will attempt to make a job with a plant a top priority')
+    
+    for _link, job in utils.listpairs(df.global.world.jobs.list) do
+        if plant.pos.x == job.pos.x
+        and plant.pos.y == job.pos.y
+        and plant.pos.z == job.pos.z
+        then
+            doJobNow(job)
+            return
+        end
+    end
+    
+    print("Couldn't find any job involving this plant.")
+end
+
+local function doWorkOrderJobsNow(order)
+    local needle = order.id
+    local cnt = 0
+    for _link, job in utils.listpairs(df.global.world.jobs.list) do
+        if job.order_id == needle then
+            doJobNow(job)
+            cnt = cnt + 1
+        end
+    end
+    
+    if cnt > 0 then
+        print("Found " .. cnt .. " jobs for this work order.")
+    else
+        print("Couldn't find any job for this work order.")
+    end
+end
+
+local function getSelectedWorkOrder(silent)
+    local scr = dfhack.gui.getCurViewscreen()
+    local orders
+    local idx
+    if df.viewscreen_jobmanagementst:is_instance(scr) then
+        orders = df.global.world.manager_orders
+        idx = scr.sel_idx
+    elseif df.viewscreen_workshop_profilest:is_instance(scr)
+        and scr.tab == df.viewscreen_workshop_profilest.T_tab.Orders
+    then
+        orders = scr.orders
+        idx = scr.order_idx
+    end
+    if orders then
+        if idx < #orders then
+            return orders[idx]
+        else
+            if not silent then qerror("Invalid work order selected") end
```

Comment:
this is dead code because silent is always true. you could make silent configurable with a commandline param, or just remove the dead code.

if you do make it configurable, you should change the `qerror` to a `dfhack.printerr`. `qerror` should only be used for errors, but `silent` implies only logging verbosity.

Reply:
well, this was mostly to make this function look the same as dfhack.gui.getSelected*-functions.

Reply:
understood, but I'd still prefer removing the dead code (you don't actually have to use the 'silent' param -- you could even remove it altogether and still pass `true` to the call. it will just get ignored)